src/lib/pubkey/dilithium/dilithium_common/dilithium_polynomials.h

```diff
@@ -0,0 +1,1609 @@
+               t[1] = static_cast<uint32_t>(mode.gamma1()) - m_coeffs[2 * i + 1];
+
+               r[5 * i + 0] = static_cast<uint8_t>(t[0]);
+               r[5 * i + 1] = static_cast<uint8_t>(t[0] >> 8);
+               r[5 * i + 2] = static_cast<uint8_t>(t[0] >> 16);
+               r[5 * i + 2] |= static_cast<uint8_t>(t[1] << 4);
+               r[5 * i + 3] = static_cast<uint8_t>(t[1] >> 4);
+               r[5 * i + 4] = static_cast<uint8_t>(t[1] >> 12);
+               }
+            }
+         }
+
+      /*************************************************
+      * Name:        polyt1_unpack
+      *
+      * Description: Unpack polynomial t1 with 10-bit coefficients.
+      *              Output coefficients are standard representatives.
+      *
+      * Arguments:   - Polynomial& r: pointer to output polynomial
+      *              - const uint8_t *a: byte array with bit-packed_t1 polynomial
+      **************************************************/
+      static void polyt1_unpack(Polynomial& r, const uint8_t* a)
+         {
+         for(size_t i = 0; i < DilithiumModeConstants::N / 4; ++i)
+            {
+            r.m_coeffs[4 * i + 0] = ((a[5 * i + 0] >> 0) | (static_cast<uint32_t>(a[5 * i + 1]) << 8)) & 0x3FF;
+            r.m_coeffs[4 * i + 1] = ((a[5 * i + 1] >> 2) | (static_cast<uint32_t>(a[5 * i + 2]) << 6)) & 0x3FF;
+            r.m_coeffs[4 * i + 2] = ((a[5 * i + 2] >> 4) | (static_cast<uint32_t>(a[5 * i + 3]) << 4)) & 0x3FF;
+            r.m_coeffs[4 * i + 3] = ((a[5 * i + 3] >> 6) | (static_cast<uint32_t>(a[5 * i + 4]) << 2)) & 0x3FF;
+            }
+         }
+
+      /*************************************************
+      * Name:        polyt1_pack
+      *
+      * Description: Bit-pack polynomial t1 with coefficients fitting in 10 bits.
+      *              Input coefficients are assumed to be standard representatives.
+      *
+      * Arguments:   - uint8_t *r: pointer to output byte array with at least
+      *                            POLYT1_PACKEDBYTES bytes
+      **************************************************/
+      void polyt1_pack(uint8_t* r) const
+         {
+         for(size_t i = 0; i < DilithiumModeConstants::N / 4; ++i)
+            {
+            r[5 * i + 0] = static_cast<uint8_t>((m_coeffs[4 * i + 0] >> 0));
+            r[5 * i + 1] = static_cast<uint8_t>((m_coeffs[4 * i + 0] >> 8) | (m_coeffs[4 * i + 1] << 2));
+            r[5 * i + 2] = static_cast<uint8_t>((m_coeffs[4 * i + 1] >> 6) | (m_coeffs[4 * i + 2] << 4));
+            r[5 * i + 3] = static_cast<uint8_t>((m_coeffs[4 * i + 2] >> 4) | (m_coeffs[4 * i + 3] << 6));
+            r[5 * i + 4] = static_cast<uint8_t>((m_coeffs[4 * i + 3] >> 2));
+            }
+         }
+
+      Polynomial() = default;
+   };
+class PolynomialVector
+   {
+   public:
+      std::vector<Polynomial> m_vec;
+
+   public:
+      // Default Consturctor inserted. Check further features for this
```

Comment:
Typo in comment

Reply:
Removed the entire comment. No idea what it should mean anyway.